compiler/noirc_evaluator/src/ssa.rs

```diff
@@ -460,9 +464,22 @@ impl SsaBuilder {
     }
 
     fn print(mut self, msg: &str) -> Self {
-        if self.print_ssa_passes {
-            self.ssa.normalize_ids();
-            println!("{msg}\n{}", self.ssa);
+        match &self.ssa_logging {
+            SsaLogging::None => (),
+            SsaLogging::All => {
+                self.ssa.normalize_ids();
+                println!("After {msg}:\n{}", self.ssa);
+            }
+            SsaLogging::Equals(string) => {
+                let string = string.to_lowercase();
+                let string = string.strip_prefix("after ").unwrap_or(&string);
+                let string = string.strip_suffix(':').unwrap_or(string);
+
+                if msg.to_lowercase().contains(string) {
+                    self.ssa.normalize_ids();
+                    println!("After {msg}:\n{}", self.ssa);
+                }
+            }
         }
```

Comment:
```suggestion
        let print_ssa_pass = match &self.ssa_logging {
            SsaLogging::None => false,
            SsaLogging::All => true,
            SsaLogging::Equals(string) => {
                let string = string.to_lowercase();
                let string = string.strip_prefix("after ").unwrap_or(&string);
                let string = string.strip_suffix(':').unwrap_or(string);
                msg.to_lowercase().contains(string)
            }
        };
        if print_ssa_pass {
            self.ssa.normalize_ids();
            println!("After {msg}:\n{}", self.ssa);
        }
```
So we only have that normalise-and-format logic once.